### PR TITLE
feat(csam): add toggle for csam scan and save info to textile

### PR DESCRIPTION
**Pull request by genie-magic**
**What this PR does** 📖
- add a toggle for can scan and save info to textile
- add textile manager for saving user consent data

**Which issue(s) this PR fixes** 🔨

AP-1332

**Special notes for reviewers** 🗒️
- This needs backend dev review too

### CODE REVIEW

**Review by Jekrimo — APPROVED** [review]

**Comment by phillsatellite** [comment]
Tested: Noticed a error in console when testing (Error does not appear in dev error message will only appear on this branch) 
I replicated this by just creating a new account and logging in. I reproduced this with csam scan toggle on and off and error appeared both times
Screen Shot 2022-04-07 at 4 51 29 PM

**Comment by genie-magic** [comment]
@phillsatellite  can you test again?

**Review by phillsatellite — APPROVED** [review]

File: store/settings/actions.ts [hunk 1]
```diff
@@ -13,4 +17,23 @@ export default {
...
+    { commit }: ActionsArguments<SettingsState>,
+    consentScan: boolean,
+  ) {
+    try {
+      commit('setConsentScan', consentScan)
+
+      const $TextileManager: TextileManager = Vue.prototype.$TextileManager
+
+      if (!$TextileManager.userInfoManager) {
+        throw new Error(TextileError.METADATA_MANAGER_NOT_FOUND)
```
Comment (pavlzk) [inline store/settings/actions.ts:30]: probably should be `throw new Error(TextileError.USERINFO_MANAGER_NOT_FOUND)`

File: libraries/Textile/UserManager.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    } catch (error) {}
+  }
+
+  /**
+   * Find the current user consent info
+   * @returns returns the current user's info
+   */
+  private async _findRecord(): Promise<UserdataFromThread | null> {
+    const query = Query.where('user_address').eq(this.textile.wallet.address)
+    const records = await this.textile.client.find<UserdataFromThread>(
```
Comment (pavlzk) [inline libraries/Textile/UserManager.ts:57]: can be simplified like this:
```
  const [record] = await this.textile.client.find<UserdataFromThread>(
      this.threadID,
      CollectionName,
      query,
    )

    return record || null
  }
```
Not important just imho :)

**Review by pavlzk — APPROVED** [review]

File: libraries/Textile/UserManager.ts (hunk 2 above)
Comment (genie-magic) [reply libraries/Textile/UserManager.ts]: @pavlzk  I updated 2 parts you pointed.

**Comment by stavares843** [comment]
Captura de ecrã 2022-04-08, às 20 21 03

can you rebase this manually please*?

**Comment by stavares843** [comment]
Captura de ecrã 2022-04-11, às 09 06 57

**Review by stavares843 — APPROVED** [review]

**Comment by genie-magic** [comment]
I am creating a new PR (https://github.com/Satellite-im/Core-PWA/pull/2809) which is rebased.
I will close this PR.

**Pull request closed without merging**